lib/components/user/monitored-trip/trip-status.js

```diff
@@ -20,9 +21,25 @@ const StyledFooterButton = styled(Button)`
 `
 
 export function FooterButton({ bsStyle, glyphIcon, onClick, text }) {
+  const [loading, setLoading] = useState(false)
+  const clickHandler = () => {
+    setLoading(true)
+    onClick()
+  }
+
+  useEffect(() => {
+    setLoading(false)
+  }, [text])
+
   return (
-    <StyledFooterButton bsStyle={bsStyle} onClick={onClick}>
-      <Glyphicon glyph={glyphIcon} /> {text}
+    <StyledFooterButton bsStyle={bsStyle} onClick={clickHandler}>
+      {loading ? (
+        <InlineLoading />
```

Comment:
It might be better to always show the text. This will prevent the button to unnecessarily resize.

Reply:
I can't replicate. It might be the `disabled` change I made.